src/Pkg/PParser.hs

```diff
@@ -79,18 +80,33 @@ filename = (do
         -- Through at least version 0.9.19.1, IPKG executable values were
         -- possibly namespaced identifiers, like foo.bar.baz.
         show <$> fst <$> iName []
-    if isValidFilename filename
+    let errorMessage = filenameErrorMessage filename
+    if isNothing errorMessage
       then return filename
-      else fail "a filename must be non-empty and have no directory component")
+      else fail $ fromJust errorMessage)
```

Comment:
This `if isNothing`/`fromJust` use looks like it should be replaced with a `case` or a call to `maybe`.